v-next/hardhat/src/internal/cli/init/package-manager.ts

```diff
@@ -0,0 +1,153 @@
+import { execSync } from "node:child_process";
+import path from "node:path";
+
+import { exists } from "@ignored/hardhat-vnext-utils/fs";
+import semver from "semver";
+
+type PackageManager = "npm" | "yarn" | "pnpm";
+
+/**
+ * getPackageManager returns the name of the package manager used in the workspace.
+ * It determines this by checking the presence of package manager specific lock files.
+ *
+ * @param workspace The path to the workspace where the package manager should be checked.
+ * @returns The name of the package manager used in the workspace.
+ */
+export async function getPackageManager(
+  workspace: string,
+): Promise<PackageManager> {
+  const pathToYarnLock = path.join(workspace, "yarn.lock");
+  const pathToPnpmLock = path.join(workspace, "pnpm-lock.yaml");
+
+  if (await exists(pathToYarnLock)) {
+    return "yarn";
+  }
+  if (await exists(pathToPnpmLock)) {
+    return "pnpm";
+  }
+  return "npm";
```

Comment:
What if instead of always defaulting to `npm` here we add a check based on `import.meta.filename`? That way we can know what was used to install Hardhat itself.

Something like:

```ts
if (import.meta.filename.includes("yarn")) {
  return "yarn";
}

if (import.meta.filename.includes("pnpm")) {
  return "pnpm";
}
```

I think we should prioritize the lockfiles, but this can help a bit.

Reply:
Won't the `import.meta.filename` just contain an absolute path to `package-manager.js`?